source/utils/types.ts

```diff
@@ -50,19 +52,34 @@ export interface InterfaceWithDefaults extends Instance {
 	};
 }
 
-interface RetryOption {
-	retries?: ((retry: number, error: Error) => number) | number;
+export type RetryFn = (retry: number, error: Error) => number;
+
+export interface RetryOption {
+	retries?: RetryFn | number;
 	methods?: Method[];
 	statusCodes?: StatusCode[];
 	maxRetryAfter?: number;
 	errorCodes?: ErrorCode[];
 }
 
+export type RequestFn = typeof https.request;
+
 export interface MergedOptions extends Options {
 	retry: RetryOption;
 }
 
-export interface Options extends RequestOptions {
+export interface Delays {
+	lookup?: number;
+	connect?: number;
+	secureConnect?: number;
+	socket?: number;
+	response?: number;
+	send?: number;
+	request?: number;
+}
+
+// The library overrides the type definition of `agent`.
+export interface Options extends Omit<https.RequestOptions, 'agent'> {
```

Comment:
Do you really have to omit `agent` here? I thought extended interfaces could override the parent one.

Reply:
I think that would work if `agent` were being overridden with a type that was more specific / restrictive than `https.RequestOptions`. And then, `Options` would still be a subtype of `http.RequestOptions`.

However, the override is more permissive, so I have to drop the old `agent` and define a new one. Instances of the new `Options` type no longer conform to `https.RequestOptions`.

Reply:
Makes sense. Thanks for explaining.